### PR TITLE
Using JDTUtils to expand the tree nodes instead of using SearchEngine

### PR DESCRIPTION
**Pull request by jdneo**

### CODE REVIEW

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/CodeLensUtils.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+* Copyright (c) 2018 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
```
Comment (yaohaizh) [inline java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/CodeLensUtils.java]: @akaroml How about the issue of EPL 2.0? Do we have any update on this part?

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/CodeLensUtils.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import org.eclipse.jdt.ls.core.internal.handlers.DocumentLifeCycleHandler;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@SuppressWarnings("restriction")
+public class CodeLensUtils {
```
Comment (yaohaizh) [inline java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/CodeLensUtils.java]: Decouple the underlying concept from the UI for flexibility.  

The CodeLens/TestExplorer are the UI elements. Here, we map the UI elements to the underlying utils, which will make the underlying fragile.  You should provide the API and let the UI decide how to call these APIs. Thus, you can decouple the UI from the underlying system.

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestExplorerUtils.java [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+@SuppressWarnings("restriction")
+public class TestExplorerUtils {
+
+    public static List<TestItem> searchTestItems(List<Object> arguments, IProgressMonitor monitor)
```
Comment (yaohaizh) [inline java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestExplorerUtils.java]: Document the arguments? Maybe this is the issue of the API problem. I need redesign the execute command parameter restriction...

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/CodeLensUtils.java (hunk 2 above)
Comment (jdneo) [reply java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/CodeLensUtils.java]: Follow up:
- Remain the TestSearchUtils there
- Extract out the methods which are used to construct the TestItem into a new file: TestItemsUtils
- Extract out the methods which are used to deal with the test frameworks into a new file: TestFrameworkUtils.

File: java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestExplorerUtils.java (hunk 3 above)
Comment (jdneo) [reply java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestExplorerUtils.java]: Added JavaDoc to link to the related class

**Pull request merged**